src/collect-all/Utilities.cs

```diff
@@ -0,0 +1,467 @@
+         string tempPath = Path.Combine(dest, subDir.Name);
+         
+         // Recursive call to copy all files in the subDir
+         CopyDir(subDir.FullName, tempPath);
+      }
+   }
+
+   public void DownloadTests()
+   {
+      Directory.CreateDirectory(this.m_testRoot);
+
+      // Find the last successful test build based on OS.
+      string url = "http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/release_windows_nt/lastSuccessfulBuild/artifact/bin/tests/tests.zip";
+      string args = $"{url} -P {this.m_testRoot}";
+         
+      // Download the zip
+      Process proc = new Process();
+      
+      proc.StartInfo.FileName = "wget";
+      proc.StartInfo.Arguments = args;
+      proc.StartInfo.RedirectStandardOutput = true;
+      proc.StartInfo.RedirectStandardError = true;
+         
+      // Throw away the output
+      proc.OutputDataReceived += (sender, arg) => { var output = arg.Data; };
+      proc.ErrorDataReceived += (sender, arg) => { var err = arg.Data; };
+      
+      if (this.m_loggingVerbose)
+      {
+         Console.WriteLine($"wget {args}");
+      }
+         
+      proc.Start();
+                  
+      proc.BeginOutputReadLine();
+      proc.BeginErrorReadLine();
+
+      proc.WaitForExit();
+
+      string zipLocation = Path.Combine(this.m_testRoot, "tests.zip");
+
+      // Go over each archive entry making sure the path contains the correct
+      // system's directory specifier.
+      using (ZipArchive archive = ZipFile.OpenRead(zipLocation))
+      {
+         foreach (ZipArchiveEntry entry in archive.Entries)
+         {
+            string entryName = this.m_os == "Windows_NT" ? entry.FullName : entry.FullName.Replace("\\", "/");
+
+            // This is a directory mkdir -p
+            if (entryName.Contains("/"))
+            {
+               Directory.CreateDirectory(Path.GetDirectoryName(Path.Combine(this.m_testRoot, entryName)));
+            }
+
+            entry.ExtractToFile(Path.Combine(this.m_testRoot, entryName));
+         }
+      }
+
+      Directory.CreateDirectory($"{this.m_coreRoot}");  
+   }
+
+   public string GetOs()
+   {
+      string os = "";
+      string runtimeInfo = RuntimeInformation.OSDescription;
+      string osSplit = runtimeInfo.Split(' ')[0];
+      
+      if (osSplit == "Darwin")
+      {
+         os = "OSX";
+      }
+
+      else if (osSplit == "Linux") 
+      {
+        os = "Linux";
+      }
+      
+      else if (osSplit == "Microsoft")
+      {
+         os = "Windows_NT";
+      }
+
+      else 
+      {
+         throw new NotImplementedException($"Unsupported OS: {osSplit}");
+      }
+
+      return os;
+   }
+   
+   private class Setting
+   {
+      public string CoreClrRepoLocation;
+      public string BuildType;
+      public string Arch;
+   }
+   
+   public bool ParseSettingFile(string location = null)
+   {
+      string path = Path.Combine(System.IO.Directory.GetCurrentDirectory(), "spmi_settings.json");
```

Comment:
We want to be careful with this. What if someone is calling your superpmi tool from a different directory than exists spmi_settings.json. You'll at the very least what to confirm that the path exists.